pkg/web/http_healthchecker.go

```diff
@@ -0,0 +1,24 @@
+package web
+
+import (
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+)
+
+type healthChecker struct {
+	logger logrus.FieldLogger
+}
+
+// healthCheck reports if the server is ready to process traffic.  It does not validate downstream dependencies.
+func (hc *healthChecker) healthCheck(w http.ResponseWriter, req *http.Request) {
+	hc.logger.Info("healthCheck")
+	_, _ = w.Write([]byte("OK"))
+}
+
+// deepCheck reports on the status of downstream dependencies.  It should be non-blocking (ie, report on the status
+// of watchdogs, rather than make roundtrips).  For now, it just reports that it's always happy.
```

Comment:
What are `watchdogs`?

Reply:
https://en.wikipedia.org/wiki/Watchdog_timer

The basic idea is if you don't pat the watchdog, the watchdog kills you.

So as an example: when we get to clustering, rather than pinging Redis every time `/deepcheck` is queried, we would "pat the watchdog" (save a timestamp) whenever we get an update from Redis about a neighboring host.  `/deepcheck` would then check the watchdog (timestamp), and if it's too old, it would report a failure.

The example isn't exactly the same, but it's close enough.